include/internal/ktls.h

```diff
@@ -96,16 +106,16 @@ static ossl_inline int ktls_enable(int fd)
  * The TLS_TX socket option changes the send/sendmsg handlers of the TCP socket.
  * If successful, then data sent using this socket will be encrypted and
  * encapsulated in TLS records using the crypto_info provided here.
+ * The TLS_RX socket option changes the recv/recvmsg handlers of the TCP socket.
+ * If successful, then data received using this socket will be decrypted,
+ * authenticated and decapsulated using the crypto_info provided here.
  */
 static ossl_inline int ktls_start(int fd,
                                   struct tls12_crypto_info_aes_gcm_128
                                   *crypto_info, size_t len, int is_tx)
 {
-    if (is_tx)
-        return setsockopt(fd, SOL_TLS, TLS_TX, crypto_info,
-                          sizeof(*crypto_info)) ? 0 : 1;
-    else
-        return 0;
+    return setsockopt(fd, SOL_TLS, is_tx ? TLS_TX : TLS_RX, crypto_info,
+                      sizeof(*crypto_info)) ? 0 : 1;
```

Comment:
when is sizeof(*crypt_info) == 0 ?

Reply:
oh, forget it, I missed the double )...

Reply:
it might be easier to read if the ? 0 : 1 was on a line of its own, what do you think?

Reply:
I think it only makes it less readable. But, i've move the crypto_info to the other line. Hopefully it is more readable now.

Reply:
I've reformatted this a bit.

Reply:
I hope this is more readable now.